hazelcast/src/main/java/com/hazelcast/map/impl/nearcache/NearCacheImpl.java

```diff
@@ -88,17 +88,35 @@ public NearCacheImpl(String mapName, NodeEngine nodeEngine) {
         this.lastCleanup = Clock.currentTimeMillis();
         this.serializationService = nodeEngine.getSerializationService();
         this.invalidateOnChange = nearCacheConfig.isInvalidateOnChange();
-        this.mapName = mapName;
+    }
+
+    private int getMaxSize() {
+        MapConfig mapConfig = mapContainer.getMapConfig();
+        NearCacheConfig nearCacheConfig = mapConfig.getNearCacheConfig();
+        int maxSize = nearCacheConfig.getMaxSize();
+        return maxSize <= 0 ? Integer.MAX_VALUE : maxSize;
+    }
+
+    private long getMaxIdleMillis() {
+        MapConfig mapConfig = mapContainer.getMapConfig();
+        NearCacheConfig nearCacheConfig = mapConfig.getNearCacheConfig();
+        return TimeUnit.SECONDS.toMillis(nearCacheConfig.getMaxIdleSeconds());
+    }
+
+    private long getTimeToLiveMillis() {
+        MapConfig mapConfig = mapContainer.getMapConfig();
+        NearCacheConfig nearCacheConfig = mapConfig.getNearCacheConfig();
+        return TimeUnit.SECONDS.toMillis(nearCacheConfig.getTimeToLiveSeconds());
     }
 
     // TODO this operation returns the given value in near-cache memory format (data or object)?
     @Override
     public void put(Data key, Object value) {
         fireTtlCleanup();
-        if (evictionPolicy == NONE && cache.size() >= maxSize) {
+        if (evictionPolicy == NONE && cache.size() >= getMaxSize()) {
```

Comment:
Why are you retrieving this value every time? It obfuscates the regular flow since the CPU needs to chew through a lot of additional code.